### ISSUE TITLE
Configurable dimensions for regridding

### ISSUE THREAD

**Issue by oloapinivad** [jhardenberg/smmregrid#3]
So far, the regrid operation works on a series of dimension which are kept fo the analysis.

```
    nd = sum([(d not in ['i', 'j', 'x', 'y', 'lon', 'lat', 'longitude', 'latitude',
                         'cell', 'cells', 'ncells', 'values', 'value']) for d in source_data.dims])

    kept_shape = list(source_data.shape[0:nd])
    kept_dims = list(source_data.dims[0:nd])
```

If dimensions are not met, this leads to weird results. We should add this as a configurable option and also produce a warning if this does not happen

**Comment by oloapinivad** [comment]
An error is introduced n #4

**Comment by oloapinivad** [comment]
Closed in #4
